plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java

```diff
@@ -13,36 +13,87 @@
  */
 package io.trino.plugin.hudi;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
 import io.trino.plugin.hive.HiveColumnHandle;
 import io.trino.spi.Page;
 import io.trino.spi.block.Block;
 import io.trino.spi.block.RunLengthEncodedBlock;
 import io.trino.spi.connector.ConnectorPageSource;
+import org.apache.hadoop.fs.Path;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 
+import static io.airlift.slice.Slices.utf8Slice;
 import static io.trino.plugin.base.util.Closables.closeAllSuppress;
+import static io.trino.plugin.hive.HiveColumnHandle.FILE_MODIFIED_TIME_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.FILE_MODIFIED_TIME_TYPE_SIGNATURE;
+import static io.trino.plugin.hive.HiveColumnHandle.FILE_SIZE_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.FILE_SIZE_TYPE_SIGNATURE;
+import static io.trino.plugin.hive.HiveColumnHandle.PARTITION_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.PARTITION_TYPE_SIGNATURE;
+import static io.trino.plugin.hive.HiveColumnHandle.PATH_COLUMN_NAME;
+import static io.trino.plugin.hive.HiveColumnHandle.PATH_TYPE;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.DateTimeEncoding.packDateTimeWithZone;
+import static io.trino.spi.type.TimeZoneKey.UTC_KEY;
 import static java.util.Objects.requireNonNull;
 
 public class HudiPageSource
         implements ConnectorPageSource
 {
-    private final List<HiveColumnHandle> columnHandles;
+    private final Block[] prefilledBlocks;
+    private final int[] delegateIndexes;
     private final ConnectorPageSource dataPageSource;
-    private final Map<String, Block> partitionBlocks;
 
     public HudiPageSource(
+            String partitionName,
             List<HiveColumnHandle> columnHandles,
             Map<String, Block> partitionBlocks,
-            ConnectorPageSource dataPageSource)
+            ConnectorPageSource dataPageSource,
+            Path path,
+            long fileSize,
```

Comment:
Just for my understanding, how does file size within a page help? Is it used in table statistics as well?

Reply:
The size exists for just providing `$file_size`. It unrelates to table statistics.